### PR TITLE
Towards pydantic dataclass

### PR DESCRIPTION
**Pull request by mgcam**

### CODE REVIEW

**Review by nerdstrike — APPROVED** [review]
I see nothing bad here. I think you've found the best way yet.

File: lang_qc/models/pacbio/well.py [hunk 1]
```diff
@@ -33,39 +33,83 @@
...
-    if any.
+    The class contains the attributes that uniquely define this well (`run_name`,
+    `label`, `plate_number`, `id_product`), along with the time line and the
+    current QC state of this well, if any.
+
+    Optionally, the model might contain information about the current QC state
+    of the well.
+
+    The best way to instantiate the object of this class is via the constructor,
+    giving it the an ORM object representing a database row with information
```
Comment (nerdstrike) [inline lang_qc/models/pacbio/well.py:68]: Minor grammar confusion.

**Review by nerdstrike — CHANGES_REQUESTED** [review]
One minor doc error to correct.

I like the approach. We could at some point extend pydantic.base_model to include the `get_field_names` method so it is usable everywhere. `pre_root` is almost general, but still knows about the sqlalchemy table it maps to. Do you think we can meaningfully have a table-mapping `pre_root` that can do the job for any sqlalchemy-single-table -> class transformation?

**Comment by mgcam** [comment]
> Do you think we can meaningfully have a table-mapping `pre_root` that can do the job for any sqlalchemy-single-table -> class transformation?

Yes, to some degree. I expect we will factor out some functionality into stand-alone functions as we convert more classes ot this type of model.

**Review by nerdstrike — APPROVED** [review]

**Pull request merged**